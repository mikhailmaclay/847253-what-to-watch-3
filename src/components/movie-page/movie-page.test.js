// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
// Components
import MoviePage from './movie-page';

const movies = [
  {
    id: 1,
    name: `The Grand Budapest Hotel`,
    genre: `Drama`,
    releaseDate: `2014`,
    description: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`
    ],
    runTime: 80,
    team: [
      {fullName: `Wes Anderson`, role: `Director`},
      {fullName: `Bill Murray`, role: `Actor`},
      {fullName: `Edward Norton`, role: `Actor`},
      {fullName: `Jude Law`, role: `Actor`},
      {fullName: `Willem Dafoe`, role: `Actor`},
      {fullName: `Adrien Brody`, role: `Actor`}
    ],
    reviews: [1],
    poster: `/img/the-grand-budapest-hotel-poster.jpg`,
    background: `/img/bg-the-grand-budapest-hotel.jpg`,
  }
];

const reviews = [
  {
    id: 1,
    user: 1,
    rating: 5,
    date: `November 18, 2020 03:24:00`,
    text: `DO! IT!`
  }
];

const users = [
  {id: 1, fullName: `Shia LaBeouf`}
];

const match = {
  params: {
    id: 1
  }
};

it(`<MoviePage/> should render correctly`, () => {
  const result = renderer.create(
      <BrowserRouter>
        <MoviePage movies={movies} reviews={reviews} users={users} match={match}/>
      </BrowserRouter>
  ).toJSON();

  expect(result).toMatchSnapshot();
});
