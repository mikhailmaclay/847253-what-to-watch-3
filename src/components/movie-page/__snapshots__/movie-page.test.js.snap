// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviePage/> Should render correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
        style={
          Object {
            "backgroundColor": "#977461",
          }
        }
      >
        <img
          alt="Shutter Island"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Shutter_Island.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link "
            href={null}
            onClick={null}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Shutter Island
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Thriller
            </span>
            <span
              className="movie-card__year"
            >
              January 01, 2010
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/films/1/player"
              onClick={[Function]}
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Shutter Island"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Shutter_Island.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="/films/1"
                  onClick={[Function]}
                  style={
                    Object {
                      "pointerEvents": "",
                      "userSelect": "none",
                    }
                  }
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="/films/1/details"
                  onClick={[Function]}
                  style={
                    Object {
                      "pointerEvents": "",
                      "userSelect": "none",
                    }
                  }
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="/films/1/reviews"
                  onClick={[Function]}
                  style={
                    Object {
                      "pointerEvents": "",
                      "userSelect": "none",
                    }
                  }
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              4.1
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Not bad, not bad...
              </span>
              <span
                className="movie-rating__count"
              >
                1002557 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.
            </p>
            <p
              className="movie-card__director"
            >
              <b>
                Director
                : 
                Martin Scorsese
              </b>
            </p>
            <p
              className="movie-card__starring"
            >
              <b>
                Starring: 
                Leonardo DiCaprio, Emily Mortimer, Mark Ruffalo
              </b>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href={null}
          onClick={null}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
